packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/tasks/TaskUtilities.java

```diff
@@ -90,7 +90,8 @@ public static TaskRequest getEnrichDatasetTaskRequest(
 		input.setLlm(llm);
 		if (document != null) {
 			try {
-				input.setDocument(objectMapper.writeValueAsString(document.getExtractions()));
+				// FIXME: Specialization for dataset enrichment?
+				input.setDocument(objectMapper.writeValueAsString(document.getExtraction().getDocumentText()));
```

Comment:
Should we add a method `getText` on the `Document` class to provide `document.getExtraction().getDocumentText()` more efficiently?

Reply:
Probably not, not in this PR anyway.
There is already a `DocumentAsset.getText()` ... which is not really being used if I am reading the code right, that field needs to be scrubbed.